docs/batch_inference_with_mms.md

```diff
@@ -0,0 +1,367 @@
+
+            img_arr = mx.image.imresize(img_arr, w, h, 2)
+            img_arr = mx.nd.transpose(img_arr, (2, 0, 1))
+            self._num_requests = idx + 1
+            img_list.append(img_arr)
+        
+        logging.debug("Worker :{} received {} requests".format(os.getpid(), self._num_requests))
+        reqs = mx.nd.stack(*img_list)
+        reqs = reqs.as_in_context(self.mxnet_ctx)
+
+        if (self._batch_size - self._num_requests) != 0:
+            padding = mx.nd.zeros((self._batch_size - self._num_requests, c, h, w), self.mxnet_ctx, 'uint8')
+            reqs = mx.nd.concat(reqs, padding, dim=0)
+
+        return reqs
+```
+**NOTE: The above code handles the case where the `handler` doesn't receive a `batch_size` number of requests. This is because MMS waits for a `max_batch_delay` amount of time to receive 
+`batch_size` number of requests. If the `max_batch_delay` timer times out before receiving `batch_size` number of requests, MMS bundles what ever requests it received
+and sends it to the handler for processing.**
+
+#### Inference logic
+The inference logic is similar to the inference logic of processing single requests. Since this isn't as interesting, we will skip explaining this in detail. Sample logic is shown below for
+completeness of this document.
+
+```python
+import mxnet as mx
+from collections import namedtuple
+
+class MXNetVisionServiceBatching(object):
+...
+    def initialize(self, context):
+        ...
+...
+    def preprocess(self, request):
+        ...
+...
+    def inference(self, model_input):
+        Batch = namedtuple('Batch', ['data'])
+        if self.error is not None:
+            return None
+
+        self.mx_model.forward(Batch([model_input]), is_train=False)
+        outputs = self.mx_model.get_outputs()
+        res = mx.ndarray.split(outputs[0], axis=0, num_outputs=outputs[0].shape[0])
+        res = [res] if not isinstance(res, list) else res
+        return res
+...    
+``` 
+
+Lets move onto post-processing. 
+
+ 
+#### Postprocess logic
+The output of inference logic is fed to the post-processing logic. As we saw before, during preprocessing, if there aren't `batch_size` number of requests, preprocessing logic
+pads the difference with 0's. These are artificially created requests sent to the module for inference. **In post-process we should ignore these artificially padded requests**. 
+Lets look at the sample logic
+
+```python
+import mxnet as mx
+from collections import namedtuple
+
+class MXNetVisionServiceBatching(object):
+...
+    def initialize(self, context):
+        ...
+...
+    def preprocess(self, request):
+        ...
+...
+    def inference(self, model_input):
+        ...
+    def postprocess(self, data):
+        res = []
+        for idx, resp in data[:self._num_requests]:
+            if idx not in self.erroneous_reqs:
+                res.append(self.top_probability(resp, self.labels, top=5))
+            else:
+                res.append("Illegal request")
```

Comment:
I suggest to update this to be more informative and include index of the erroneous request as well as different message. For example: "This request was not processed successfully. Refer to mms.log for additional information".

Reply:
Will do.